src/test/suite/recorded.test.ts

```diff
@@ -0,0 +1,95 @@
+import * as assert from "assert";
+import { promises as fsp } from "fs";
+import * as path from "path";
+import * as yaml from "js-yaml";
+import * as vscode from "vscode";
+import NavigationMap from "../../NavigationMap";
+import TestCase, {
+  SerializedPosition,
+  SerializedRange,
+  SerializedSelection,
+  TestCaseFixture,
+} from "../../TestCase";
+import { Token } from "../../Types";
+
+function deserializePosition(position: SerializedPosition) {
+  return new vscode.Position(position.line, position.character);
+}
+
+function deserializeRange(range: SerializedRange) {
+  return new vscode.Range(
+    deserializePosition(range.start),
+    deserializePosition(range.end)
+  );
+}
+
+function deserializeSelection(
+  selection: SerializedSelection
+): vscode.Selection {
+  const active = deserializePosition(selection.active);
+  const anchor = deserializePosition(selection.anchor);
+  return new vscode.Selection(anchor, active);
+}
+
+suite("recorded test cases", async function () {
+  const directory = path.join(
+    __dirname,
+    // TODO What's the best way to handle this?
+    "../../../src/test/suite/recordings"
+  );
```

Comment:
not saying this is better or worse than what you have here, but [here's](https://github.com/microsoft/vscode-extension-samples/blob/63d86e3a0e459b2adf26487204891f7af62898ce/lsp-sample/client/src/test/helper.ts) one approach that I had bookmarked in case helpful